src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -246,7 +249,11 @@ export class ObjectExplorerService implements IObjectExplorerService {
 				nodeStatus.expandEmitter.fire(expandResponse);
 			}
 		}
+
 		if (!foundSession) {
+			// Just fire reponse with empty nodes for example: request from standalone SQL instance
```

Comment:
Not sure what this comment means here. In the original MSSQL extension it was because the NodeProvider there was only meant to handle HDFS nodes so I believe this had to do with that. But I don't think it belongs here (unless there's some reason why "standalone SQL instances" wouldn't have a session?)

Reply:
I brought it over from here, as here too when this usecase occurs, we're looking at a OE session failure (captured [here](https://github.com/microsoft/azuredatastudio/pull/21590/files#diff-65d7c0b8cc76379cf757c90842b2a5cd45ca08e1a88411c44f71c46e2d8987b1R269)) with no error info > leads to session created with no nodes.
 https://github.com/microsoft/azuredatastudio/blob/bb1f5bfffe60156d4b4407fb450a4b466e66ed95/extensions/mssql/src/objectExplorerNodeProvider/objectExplorerNodeProvider.ts#L78-L81

Reply:
We need to figure out why a node provider returning a value matters though. As I've mentioned in the past node providers should be completely optional - so if there aren't any registered then it should still complete (the ObjectExplorerProvider is the main one we care about).

Now of course there may be a bug there where that isn't true, but if that's the case then that's the bug we should be looking at fixing here instead.